### ISSUE TITLE
coredns solution

### ISSUE THREAD

**Issue by ghost** [dogbutcat/openwrt-packages#3]
https://github.com/missdeer/coredns_custom_build
coredns_custom_build or mosdns ，which IS better？

**Comment by dogbutcat** [comment]
Thank you for asking. I'm not familar with custom build coredns

What I'm care about is
  1.  Package Size. Using it on newifi3 for firmware package size limit.
  2. Query Forward. As you know, most sites that need tunnel helping have area adaptive visit IP which is most friendly to your VPS or VPN provider's IP. According to this, I need the dns record better should query from or through VPS or VPN, as the return result should be friendly to your tunnel(if the site is not deploy with cdn, that's not matter).
  3. Query Speed. This is less concern then tips above. Query going through tcp or other tcp based...eeee...not too fast.

In short, which is better should adapt to what environment you want use or need to.

**Comment by ghost** [comment]
Hi, I still want to ask, how does the "Openclash" you mentioned solve the dns pollution problem?

**Comment by ghost** [comment]
mos-chinadns is indeed relatively unstable, it seems that it is too big, compared to smartdns

**Comment by dogbutcat** [comment]
Openclash is not main for dns pollution, it is for server based fast ip query, using socks port opened by application to forward dot/doh request to upper stream dns.
I have to admit mos-chinadns is not quite stable. But I don't remember if smartdns supports socks server forward. If you only need fast dns query or dns forward, smartdns is good option as written in C/C++.

**Comment by ghost** [comment]
According to the DNS concept you provided, mainland China is the fastest + other through socks proxy, I tried smartdns and Helloworld to restore stable functions to replace mos-chinadns, smartdns chose udp local DNS, helloworld chose 443 dns2socks node, the effect can be equivalent. I hope you can give some advice.
